include/kitty/partial_truth_table.hpp

```diff
@@ -0,0 +1,213 @@
+/* kitty: C++ truth table library
+ * Copyright (C) 2017-2019  EPFL
+ *
+ * Permission is hereby granted, free of charge, to any person
+ * obtaining a copy of this software and associated documentation
+ * files (the "Software"), to deal in the Software without
+ * restriction, including without limitation the rights to use,
+ * copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the
+ * Software is furnished to do so, subject to the following
+ * conditions:
+ *
+ * The above copyright notice and this permission notice shall be
+ * included in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES
+ * OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+ * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,
+ * WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+ * OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+/*!
+  \file partial_truth_table.hpp
+  \brief Implements partial_truth_table
+
+  \author Siang-Yun Lee
+*/
+
+#pragma once
+
+#include <cstdint>
+#include <type_traits>
+#include <vector>
+#include <iostream>
+#include <iomanip>
+
+#include "detail/constants.hpp"
+#include "traits.hpp"
+
+namespace kitty
+{
+
+/*! Truth table with resizable number of bits
+*/
+struct partial_truth_table
+{
+  /*! Standard constructor.
+    \param num_bits Number of bits in use initially
+  */
+  explicit partial_truth_table( int num_bits )
```

Comment:
```suggestion
  explicit partial_truth_table( uint32_t num_bits )
```

Also make sure to merge the most recent changes from master. Dynamic and static truth tables now use `uint32_t` for `num_vars()` and `num_bits()`.